src/main/java/net/minecraftforge/common/ToolAction.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@SuppressWarnings("ClassCanBeRecord")
+public class ToolAction
+{
+    private static final Map<String, ToolAction> actions = new ConcurrentHashMap<>();
```

Comment:
Could it be possible to obtain a view of this map?
I'm thinking of tweaking-type mods or multitool mods that may want to provide a list of all available actions in a certain game instance so that users and pack makers can have a clearer picture and maybe reassign or change actions on items.

Reply:
Done.